app/routes/circulars._archive._index.tsx

```diff
@@ -291,7 +328,89 @@ export default function () {
           </Button>
         </Form>
         {featureCircularsFilterByDate && (
-          <DateSelectorButton startDate={startDate} endDate={endDate} />
+          <>
+            <DateSelectorButton
+              startDate={startDate}
+              endDate={endDate}
+              onClick={() => {
+                setShowContent((shown) => !shown)
+                setShowDateRange(false)
+              }}
+              expanded={showContent}
+            />
+            {showContent && (
+              <DetailsDropdownContent className="maxw-card-xlg">
+                <CardBody>
+                  <Grid row>
+                    {Object.entries(dateSelectorLabels)
+                      .filter(([value]) => value !== 'undefined') // don't include the undefined option as a button
+                      .map(([value, label]) => ({
+                        id: `radio-${value}`,
+                        value,
+                        label,
+                      }))
+                      .map((option) => (
+                        <Grid col={4} key={option.id}>
+                          <Radio
+                            id={option.id}
+                            name="radio-date"
+                            value={option.value}
+                            label={option.label}
+                            checked={option.value === inputDateGte}
+                            defaultChecked={option.value === 'alltime'}
+                            onChange={(e) => {
+                              if (option.value === 'custom') {
+                                setShowDateRange(e.target.checked)
+                              } else {
+                                setFuzzyTime(option.value)
+                              }
+                            }}
+                          />
+                        </Grid>
+                      ))}
```

Comment:
Combine the two `.map` calls into one:

```suggestion
                      .map(([value, label]) => (
                        <Grid col={4} key={value}>
                          <Radio
                            id={`radio-${value}`}
                            name="radio-date"
                            value={value}
                            label={label}
                            checked={value === inputDateGte}
                            defaultChecked={value === 'alltime'}
                            onChange={(e) => {
                              if (value === 'custom') {
                                setShowDateRange(e.target.checked)
                              } else {
                                setFuzzyTime(option.value)
                              }
                            }}
                          />
                        </Grid>
                      ))}
```

Reply:
resolved in [3c05f3c](https://github.com/nasa-gcn/gcn.nasa.gov/pull/1482/commits/3c05f3cfd8ebc1718968e58f7c180ab0bb09bf67)